content/docs/rfcs/17/README.md

```diff
@@ -2,127 +2,136 @@
-`authPath` is  the information by which one can prove its membership in the group. <!-- TODO explain what is atuh path -->
-To construct `authPath`, peers need to locally store a Merkle tree out of the group members public keys. 
-Peers need to keep the tree updated with the recent state of the group.  
-Further inputs to the proof generation which are public are tree's `root`, `epoch` and `payload||contentTopic`  where `payload` and `contentTopic` come from the `WakuMessage`. 
-The tree `root` can be obtained from the locally maintained Merkle tree.
-The proof generation results in the following data items which are encoded inside the `proof`:  
-1. `share_x`
-2. `share_y`
-3. `nullifier`
-4. `zkSNARKs`
-
-The preceding values as well as the tree `root` (based on which the proof is generated) are encoded inside the `proof` as `|zkSNARKs<256>|root<32>|epoch<32>|share_x<32>|share_y<32>|nullifier<32>|`.
-The numbers enclosed in angle brackets indicate the bit length of the corresponding data item.
-The tuple of (`nullifier`, `share_x`, `share_y`)  can be seen as partial disclosure of peer's `sk` for the intended `epoch`.
-Given two such tuples with identical `nullifier` but distinct `share_x`, `share_y` results in full disclosure of peer's `sk` and hence burning the associated deposit.
-Note that the `nullifier` is a deterministic value derived from `sk` and `epoch` therefore any two messages issued by the same peer (i.e., sing the same `sk`) for the same `epoch` are guaranteed to have identical `nullifier`s.
-
-Note that the `authPath` of each peer depends on the current status of the registration tree (hence changes when new peers register).
-As such, it is recommended (and necessary for anonymity) that the publisher updates her `authPath` based on the latest status of the group and attempts the proof using her updated `authPath`.
 
 
-## Routing
+## Group  Synchronization
 
-Upon the receipt of a PubSub message, the routing peer needs to extract and parse the `proof` from the `data` field.  
-If the `epoch` attached to the message has a non-reasonable gap (TODO: the gap should be defined) with the routing peer's current `epoch` then the message must be dropped (this is to prevent a newly registered peer spamming the system by messaging for all the past epochs). 
-Furthermore, the routing peers MUST check whether the `proof` is valid and the message is not spam. 
-If both checks are passed successfully, then the message is relayed. 
-If `proof` is invalid then the message is dropped. 
-If spamming is detected, the publishing peer gets slashed. 
-An overview of routing procedure is depicted in Figure 2.
+Proof generation relies on the knowledge of Merkle tree root `merkle_root` and `authPath` which both require access to the membership Merkle tree. 
+Getting access to the Merkle tree can be done in various ways. 
+One way is that all the peers construct the tree locally.
```

Comment:
Below paragraph should be together same

Reply:
Can you please clarify this? do you mean to combine the two paragraphs?

Reply:
Yes